### DISCUSSION TITLE
Questions about global variables and policies

### DISCUSSION THREAD

**Discussion by quank123wip** [geldata/gel#4264] [Q&A]
I'm a bit of confused that the global variables is static? Is this a variable for each db connection? If I'm not getting wrong, I can't put it in use if I'm making a backend application because it's using one connection for handling all queries, is it right?

**Comment by elprans** [comment]
Global variables are dynamic.  You can change the value of a global variable at any time either via the `SET GLOBAL` command, or, normally, with client binding API.  Here's a TypeScript example:


```typescript
import createClient from 'edgedb';

const client = createClient();

const myApiHandler = async (userId: string) => {
  const scopedClient = client.withGlobals({
    current_user: userId,
  });

  return await scopedClient.query(
    `select global current_user;`
  );
}
```

**Reply by quank123wip** [reply]
> Global variables are dynamic.  You can change the value of a global variable at any time either via the `SET GLOBAL` command, or, normally, with client binding API.  Here's a TypeScript example:
> 
> 
> ```typescript
> import createClient from 'edgedb';
> 
> const client = createClient();
> 
> const myApiHandler = async (userId: string) => {
>   const scopedClient = client.withGlobals({
>     current_user: userId,
>   });
> 
>   return await scopedClient.query(
>     `select global current_user;`
>   );
> }
> ```

So if I'm handling requests from different users, I need change it every time, right?

**Reply by elprans** [reply]
Yes.  To clarify, you're not actually literally using a single connection.  `createClient` returns a _Client_ which manages connections intelligently.  With EdgeDB you don't need to worry about individual connections, just create a single client and (re-)configure it when needed as shown above.

**Reply by quank123wip** [reply]
> Yes.  To clarify, you're not actually literally using a single connection.  `createClient` returns a _Client_ which manages connections intelligently.  With EdgeDB you don't need to worry about individual connections, just create a single client and (re-)configure it when needed as shown above.

Thanks for your help, I have understood this. Further than this question, do you have plan to make policies can be used without global variables? When I dive into a traditional backend project, of course I can't create clients for every user, but still I have to deal with permissions. Then I have to make new clients every queries or grab all datas to do it in backend(and perhaps slow). More than this, there might be recursive policies that makes me have to make recursive request in backend. I think a optional solution without global variables would be helpful? After all not everyone would use a raw database as their backend right?

**Reply by elprans** [reply]
I'm not sure I understand your request.  You don't _have_ to use globals in your access policies, it's just a convenient way of communicating context, such as the current user id.
